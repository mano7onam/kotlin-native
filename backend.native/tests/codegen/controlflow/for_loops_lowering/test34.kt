package codegen.controlflow.for_loops_lowering.test34

import kotlin.test.*

@Test fun runTest() {
  val arr = arrayOf(4, 5, 6, 14, 15, 16, 24, 25, 26, 34, 35, 36, 0, 50, 60, 70)
  fun foo() : Int = 10
  fun bar() : Int { println("bar"); return 14 }
  for (i in arr) {
    if (i in arrayOf(14, 24, 24)) {
      println(i)
    }
    if (i in 0 .. foo()) {
      println(i)
    }
    if (i in 0 until bar()) {
      println(i)
    } 
    if (i in 15 downTo 0) {
      println(i)
    }
    if (i in arrayOf(1, 2, 3, 4, 5, 6, 7, 8, 9, 10).indices) {
      println(i)
    }
  }
}
