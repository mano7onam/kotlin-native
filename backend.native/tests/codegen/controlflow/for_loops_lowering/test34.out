4
bar
4
4
4
5
bar
5
5
5
6
bar
6
6
6
14
bar
14
bar
15
bar
24
bar
bar
bar
bar
bar
bar
0
bar
0
0
0
bar
bar
bar
